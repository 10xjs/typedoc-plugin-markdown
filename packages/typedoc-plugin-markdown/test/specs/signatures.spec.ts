import * as Handlebars from 'handlebars';
import { SignatureReflection } from 'typedoc';

import { TestApp } from '../test-app';

describe(`Signatures:`, () => {
  let testApp: TestApp;
  let partial: Handlebars.TemplateDelegate;
  beforeAll(() => {
    testApp = new TestApp(['signatures.ts']);
    testApp.bootstrap();
    TestApp.stubPartials(['comment', 'member.sources']);
    partial = TestApp.getPartial('member.signature');
  });

  test(`should compile callable signature'`, () => {
    expect(
      TestApp.compileTemplate(
        partial,
        testApp.findReflection('CallableSignature').signatures[0],
      ),
    ).toMatchSnapshot();
  });

  test(`should compile signature with params'`, () => {
    expect(
      TestApp.compileTemplate(
        partial,
        testApp.findReflection('functionWithParameters')
          .signatures[0] as SignatureReflection,
      ),
    ).toMatchSnapshot();
  });

  test(`should compile signature with rest params'`, () => {
    expect(
      TestApp.compileTemplate(
        partial,
        testApp.findReflection('functionWithRest')
          .signatures[0] as SignatureReflection,
      ),
    ).toMatchSnapshot();
  });

  test(`should compile signature with optional params'`, () => {
    expect(
      TestApp.compileTemplate(
        partial,
        testApp.findReflection('functionWithOptionalParam')
          .signatures[0] as SignatureReflection,
      ),
    ).toMatchSnapshot();
  });

  test(`should compile signature with union types'`, () => {
    expect(
      TestApp.compileTemplate(
        partial,
        testApp.findReflection('functionWithUnionTypes')
          .signatures[0] as SignatureReflection,
      ),
    ).toMatchSnapshot();
  });

  test(`should compile signature with default values'`, () => {
    expect(
      TestApp.compileTemplate(
        partial,
        testApp.findReflection('functionWithUnionTypes')
          .signatures[0] as SignatureReflection,
      ),
    ).toMatchSnapshot();
  });
});
