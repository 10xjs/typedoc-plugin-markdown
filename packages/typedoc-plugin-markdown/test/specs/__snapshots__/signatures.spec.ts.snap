// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signatures: should compile callable signature' 1`] = `
"▸ (): *string*

[partial: comment]

[partial: member.sources]
**Returns:** *string*
"
`;

exports[`Signatures: should compile signature with default values' 1`] = `
"▸ **functionWithUnionTypes**(\`arg\`: boolean[] | number, ...\`args\`: (string | number)[]): *any*

[partial: comment]

#### Parameters:

Name | Type |
------ | ------ |
\`arg\` | boolean[] &#124; number |
\`...args\` | (string &#124; number)[] |

[partial: member.sources]
**Returns:** *any*
"
`;

exports[`Signatures: should compile signature with optional params' 1`] = `
"▸ **functionWithOptionalParam**(\`requiredParam\`: string, \`optionalParam?\`: string): *void*

[partial: comment]

#### Parameters:

Name | Type | Description |
------ | ------ | ------ |
\`requiredParam\` | string | A normal parameter. |
\`optionalParam?\` | string | An optional parameter.  |

[partial: member.sources]
**Returns:** *void*
"
`;

exports[`Signatures: should compile signature with params' 1`] = `
"▸ **functionWithParameters**(\`paramZ\`: string, \`paramG\`: any, \`paramA\`: any): *number*

[partial: comment]

#### Parameters:

Name | Type | Description |
------ | ------ | ------ |
\`paramZ\` | string | This is a string parameter. |
\`paramG\` | any | This is a parameter flagged with any.     This sentence is placed in the next line.  |
\`paramA\` | any |    This is a **parameter** pointing to an interface.   |

[partial: member.sources]
**Returns:** *number*
"
`;

exports[`Signatures: should compile signature with rest params' 1`] = `
"▸ **functionWithRest**(...\`rest\`: string[]): *string*

[partial: comment]

#### Parameters:

Name | Type | Description |
------ | ------ | ------ |
\`...rest\` | string[] | The rest parameter.  |

[partial: member.sources]
**Returns:** *string*
"
`;

exports[`Signatures: should compile signature with union types' 1`] = `
"▸ **functionWithUnionTypes**(\`arg\`: boolean[] | number, ...\`args\`: (string | number)[]): *any*

[partial: comment]

#### Parameters:

Name | Type |
------ | ------ |
\`arg\` | boolean[] &#124; number |
\`...args\` | (string &#124; number)[] |

[partial: member.sources]
**Returns:** *any*
"
`;
